algorithms/linfa-nn/src/distance.rs

```diff
@@ -177,6 +221,28 @@ mod test {
         dist_test(LInfDist, 3.9);
     }
 
+    #[test]
+    fn l0_dist_properties() {
+        // This norm does not accomplish with infinite distance test or absolute
+        // homogeneity property
+
+        let dist = L0Dist;
+        let a = arr1(&[0.5, 6.6]);
+        let b = arr1(&[4.4, 3.0]);
+        assert_eq!(dist.distance(a.view(), b.view()), 2.0);
+
+        test_triangle_inequality(&dist);
+        test_symmetry(&dist);
+    }
+
+    #[test]
+    #[should_panic]
+    fn l0_dist_non_properties() {
+        let dist = L0Dist;
+        test_infinite_distance(&dist);
```

Comment:
conflicting with homogeneity is actually never tested when infinite distance already panics :D